### PR TITLE
Admin Dashboard: Read in seed data

### PR DESCRIPTION
**Pull request by shannonmhong**
# Admin Dashboard: Reading and Populating User Dashboard with Seed Data

Completed Features
- Read seed data - successfully read from database and populate with users
Note: Only displayed one page but made functionality for specifying what page of the data you want to see, and displaying the correct one (in this case I only displayed page 0. will implement pagination next sprint).

(styling is not perfect right now)

An error shows up for me when I attempt to use query. I am assuming I can fix this by putting the routing method for `findMany` that is currently under index in a different file and rename it something like `[pageNumber].tsx` Will add in next time! Another note is that my paging logic right now hardcodes `USER_PAGE_SIZE` from `findMany`. Wondering if we should create a file with constants so it can accessed from this file.

EDIT: Error fixed! Removed pagination logic and the query statement so its working and good to go. However will run into issue again in the future when trying to implement pagination logic

image

## How to Test the feature
1) run `yarn dev`
2) login
3) go to `.../admin/users`

## PR Checklist

Does your branch (check if true):
[X] work when you run `yarn build`?
[X] successfully run `yarn:db-migrate up` without yielding any errors?
[X] successfully run `yarn prisma introspect` without yielding any errors?
[X] successfully run `yarn dev` and support all changes that your branch intends to perform?

## Related PRs

N/A

## Migrations

None

## Screenshots
image

CC: @erinysong

### CODE REVIEW

File: pages/admin/users.tsx [hunk 1]
```diff
@@ -1,5 +1,95 @@
...
+
+const UserDashboard: React.FunctionComponent = () => {
+  const [users, setUsers] = useState<User[]>();
+  const [index, setIndex] = useState(0);
+
+  const getUsers = async (pageNumber: number): Promise<void> => {
+    try {
+      const response = await fetch("http://localhost:3000/api/admin/users", {
+        method: "GET",
+        query: pageNumber,
```
Comment (sydbui) [inline pages/admin/users.tsx:57]: This line throws the error "Object literal may only specify known properties, and 'query' does not exist in type 'RequestInit' " when running yarn build

**Review by sydbui — CHANGES_REQUESTED** [review]
hi! Besides the yarn build error, everything looks pretty good to me :)

**Comment by erinysong** [comment]
good progress - this is a really good foundation for the users dashboard! let's not merge this into our branch until we fix the build error. good intuition on making a constants file we can add page size to - we'll probably end up adding it as a global variable so we can access it anywhere we want to call any of the page methods

File: pages/admin/users.tsx [hunk 2]
```diff
@@ -1,5 +1,95 @@
 import DashboardLayout from "components/DashboardLayout";
-import { useState } from "react";
+import { useEffect, useState } from "react";
+
+const UI_PAGE_SIZE = 7;
+const BACKEND_PAGE_SIZE = 21;
+
+const getBackendPageNumber = (uiPage: number): number[] => {
```
Comment (erinysong) [inline pages/admin/users.tsx:7]: can we have a comment here giving context on what getBackendPageNumber is done and used for?

File: pages/admin/users.tsx [hunk 3]
```diff
@@ -1,5 +1,95 @@
...
+};
+
+interface User {
+  name: string;
+  email: string;
+  image: string;
+  phoneNumber: string;
+}
+
+const UserDashboardItem: React.FunctionComponent<User> = ({
```
Comment (erinysong) [inline pages/admin/users.tsx:20]: great job with the dashboard and dashboard components! we're hoping to reuse these components across different parts of the app, so would it be possible to 1) turn these into components and move them to the /components folder and 2) use these components in the users page?

**Review by erinysong — CHANGES_REQUESTED** [review]
great work you made a lot of really awesome progress on the dashboard! left a few comments on relocating some components that you used to the components folder + documentation but other than that things should be all right. let's hold off merging this to main until we figure out the build error

File: pages/admin/users.tsx (hunk 2 above)
Comment (shannonmhong) [reply pages/admin/users.tsx:7]: will delete this part for now to fix build issues

**Comment by shannonmhong** [comment]
Made all suggested changes! Removed query and pagination logic to fix build errors

**Comment by erinysong** [comment]
good progress - this is a really good foundation for the users dashboard! let's not merge this into our branch until we fix the build error. good intuition on making a constants file we can add page size to - we'll probably end up adding it as a global variable so we can access it anywhere we want to call any of the page methods

**Pull request merged**